src/node/mod.rs

```diff
@@ -0,0 +1,119 @@
+use crate::color::Color;
+use crate::NodeId;
+use crate::PixelCoordinate;
+use crate::PixelDimension;
+use ambassador::{delegatable_trait, Delegate};
+use serde::{Deserialize, Serialize};
+
+mod file;
+mod group;
+mod link;
+mod text;
+
+pub use file::FileNode;
+pub use group::{Background, BackgroundStyle, GroupNode};
+pub use link::LinkNode;
+pub use text::TextNode;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct GenericNode {
+    pub id: String,
+    x: PixelCoordinate,
+    y: PixelCoordinate,
+    width: PixelDimension,
+    height: PixelDimension,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    color: Option<crate::color::Color>,
+}
+
+impl GenericNode {
+    pub fn new(
+        id: String,
+        x: PixelCoordinate,
+        y: PixelCoordinate,
+        width: PixelDimension,
+        height: PixelDimension,
+        color: Option<Color>,
+    ) -> Self {
+        Self {
+            id,
+            x,
+            y,
+            width,
+            height,
+            color,
+        }
+    }
+}
+
+#[delegatable_trait]
+pub trait GenericNodeInfo {
+    fn id(&self) -> &crate::NodeId;
+    fn get_x(&self) -> PixelCoordinate;
+    fn get_y(&self) -> PixelCoordinate;
+    fn get_width(&self) -> PixelDimension;
+    fn get_height(&self) -> PixelDimension;
+    fn color(&self) -> &Option<Color>;
+}
+
+pub use ambassador_impl_GenericNodeInfo;
```

Comment:
Ok, works better with pub use after the declaration :)

Reply:
Added a comment for it too, for the next person!